src/transport/SessionHolder.cpp

```diff
@@ -19,6 +19,11 @@
 
 namespace chip {
 
+SessionHolder::SessionHolder(const SessionHandle & session) : mSession(InPlace, session.mSession)
+{
+    session->AddHolder(*this);
```

Comment:
Why am I allowed to call a mutator on a const object? is this really intentional?

Could we make SessionHandles be const-correct? The ability to have operator-> return a non-const that you can modify seems a bit odd. Is this intentional?

Reply:
The const qualifies the SessionHandle, not the containing session, the session is still mutable. So it is intentional.